[Doc] update pull_request_template.md (#6)
GigaxGames/gigax: pull_request_template.md

<h3>PR Title and Classification</h3>
<p>Only specific types of PRs will be reviewed. The PR title is prefixed appropriately to indicate the type of change. Please use one of the following:</p>
<ul>
    <li><code>[Bugfix]</code> for bug fixes.</li>
    <li><code>[CI/Build]</code> for build or continuous integration improvements.</li>
    <li><code>[Doc]</code> for documentation fixes and improvements.</li>
    <li><code>[Model]</code> for adding a new model or improving an existing model. Model name should appear in the title.</li>
    <li><code>[Core]</code> for changes in the core gigax logic</li>
    <li><code>[Misc]</code> for PRs that do not fit the above categories. Please use this sparingly.</li>
</ul>
<p><strong>Note:</strong> If the PR spans more than one category, please include all relevant prefixes.</p>

<h3>Code Quality</h3>

<p>The PR need to meet the following code quality standards:</p>

<ul>
    <li>The code needs to be well-documented to ensure future contributors can easily understand the code.</li>

>     <li>Include sufficient tests to ensure the project to stay correct and robust: these should be added to the `./tests` folder in the rpo.</li>

Reviewer: ```suggestion
    <li>Include sufficient tests to ensure the project to stay correct and robust: these should be added to the <code><code>./tests</code> folder in the rpo.</li>
```

<br>

<p>Thank you for your contribution to gigax! Before submitting the pull request, please ensure the PR meets the following criteria. This helps gigax maintain the code quality and improve the efficiency of the review process.</p>

<h3>PR Title and Classification</h3>
<p>Only specific types of PRs will be reviewed. The PR title is prefixed appropriately to indicate the type of change. Please use one of the following:</p>
<ul>
    <li><code>[Bugfix]</code> for bug fixes.</li>
    <li><code>[CI/Build]</code> for build or continuous integration improvements.</li>
    <li><code>[Doc]</code> for documentation fixes and improvements.</li>
    <li><code>[Model]</code> for adding a new model or improving an existing model. Model name should appear in the title.</li>
    <li><code>[Core]</code> for changes in the core gigax logic</li>
    <li><code>[Misc]</code> for PRs that do not fit the above categories. Please use this sparingly.</li>
</ul>
<p><strong>Note:</strong> If the PR spans more than one category, please include all relevant prefixes.</p>

<h3>Code Quality</h3>

<p>The PR need to meet the following code quality standards:</p>

<ul>

>     <li>We adhere to <a href="https://google.github.io/styleguide/pyguide.html">Google Python style guide</a>.</li>
>     <li>The code need to be well-documented to ensure future contributors can easily understand the code.</li>
>     <li>Include sufficient tests to ensure the project to stay correct and robust. This includes both unit tests and integration tests.</li>

Reviewer: ```suggestion
    <li>The code needs to be well-documented to ensure future contributors can easily understand the code.</li>
    <li>Include sufficient tests to ensure the project to stay correct and robust: these should be added to the `./tests` folder in the rpo.</li>
```

FILL IN THE PR DESCRIPTION HERE

FIX #xxxx (*link existing issues this PR will resolve*)

**BEFORE SUBMITTING, PLEASE READ THE CHECKLIST BELOW AND FILL IN THE DESCRIPTION ABOVE**

---

<details>
<!-- inside this <details> section, markdown rendering does not work, so we use raw html here. -->
<summary><b> PR Checklist (Click to Expand) </b></summary>

<br>

<p>Thank you for your contribution to gigax! Before submitting the pull request, please ensure the PR meets the following criteria. This helps gigax maintain the code quality and improve the efficiency of the review process.</p>

<h3>PR Title and Classification</h3>
<p>Only specific types of PRs will be reviewed. The PR title is prefixed appropriately to indicate the type of change. Please use one of the following:</p>
<ul>
    <li><code>[Bugfix]</code> for bug fixes.</li>
    <li><code>[CI/Build]</code> for build or continuous integration improvements.</li>
    <li><code>[Doc]</code> for documentation fixes and improvements.</li>
    <li><code>[Model]</code> for adding a new model or improving an existing model. Model name should appear in the title.</li>
    <li><code>[Core]</code> for changes in the core gigax logic</li>

>     <li><code>[Misc]</code> for PRs that do not fit the above categories. Please use this sparingly.</li>

Reviewer: 👍

</ul>
<p><strong>Note:</strong> If the PR spans more than one category, please include all relevant prefixes.</p>

<h3>Code Quality</h3>

<p>The PR need to meet the following code quality standards:</p>

<ul>

>     <li>We adhere to <a href="https://google.github.io/styleguide/pyguide.html">Google Python style guide</a> and <a href="https://google.github.io/styleguide/cppguide.html">Google C++ style guide</a>.</li>

Reviewer: Google says in this guide:
```
Run pylint over your code using this [pylintrc](https://google.github.io/styleguide/pylintrc).
```
We should add it in CI soon 👀

>     <li>We adhere to <a href="https://google.github.io/styleguide/pyguide.html">Google Python style guide</a> and <a href="https://google.github.io/styleguide/cppguide.html">Google C++ style guide</a>.</li>

Author: https://linear.app/gigax/project/llms-v3-0e552495ed30/issues

    <li>The code need to be well-documented to ensure future contributors can easily understand the code.</li>
    <li>Include sufficient tests to ensure the project to stay correct and robust. This includes both unit tests and integration tests.</li>

>     <li>Please add documentation to <code>docs/source/</code> if the PR modifies the user-facing behaviors of gigax. It helps gigax user understand and utilize the new features or changes.</li>

Reviewer: No `docs/source` yet, because our docs are on the docusaurus frontend; but it would be great to add the docs directly to the repo here! This would mean: API docs generated automatically from GitHub code, whenever we push to this repo :) 
This is what vllm are doing with read-the-docs; and Outlines with [mkdocs (material theme)](https://squidfunk.github.io/mkdocs-material/). 
What do you think?
```suggestion
```

Author: Ok, great idea, how we can organize ourselves to achieve this.

Reviewer: I'll do it whenever I have the time :) 
[GIG-185 : Replace docusaurus with Mkdocs page](https://linear.app/gigax/issue/GIG-185/replace-docusaurus-with-mkdocs-page)

</ul>

<h3>Notes for Large Changes</h3>
<p>Please keep the changes as concise as possible. For major architectural changes, we would expect a GitHub issue (RFC) discussing the technical design and justification. Otherwise, we will tag it with <code>rfc-required</code> and might not go through the PR.</p>

<h3>What to Expect for the Reviews</h3>

<p>The goal of the gigax team is to be a <i>transparent reviewing machine</i>. We would like to make the review process transparent and efficient and make sure no contributor feel confused or frustrated. However, the gigax team is small, so we need to prioritize some PRs over others. Here is what you can expect from the review process: </p>

<ul>
    <li> After the PR is submitted, the PR will be assigned to a reviewer. Every reviewer will pick up the PRs based on their expertise and availability.</li>
    <li> After the PR is assigned, the reviewer will provide status update every 2-3 days. If the PR is not reviewed within 7 days, please feel free to ping the reviewer or the gigax team.</li>

>     <li> After the review, the reviewer will put an <code> action-required</code> label on the PR if there are changes required. The contributor should address the comments and ping the reviewer to re-review the PR.</li>

Reviewer: ```suggestion
```
Is this necessary? Maybe we can just keep doing "request changes" when submitting our review
